libs/s25main/world/GameWorldView.cpp

```diff
@@ -462,12 +462,25 @@ void GameWorldView::DrawMovingFiguresFromBelow(const TerrainRenderer& terrainRen
     }
 }
 
+constexpr auto getBqImgs()
+{
+    helpers::EnumArray<unsigned, BuildingQuality> imgs{};
+    imgs[BuildingQuality::Flag] = 50;
+    imgs[BuildingQuality::Hut] = 51;
+    imgs[BuildingQuality::House] = 52;
+    imgs[BuildingQuality::Castle] = 53;
+    imgs[BuildingQuality::Mine] = 54;
```

Comment:
shouldn't we set these in the order of the enum instead?

Reply:
I don't see a need for this. I changed the order of the enum to fit our needs: Better performance: https://github.com/Return-To-The-Roots/s25client/pull/1476/commits/0c474e86fc172214f95f3d621dca3b1dd88e9eed

Due to the `constexpr` this whole array is a constant and the function makes it very clear which BQ is which image. I find that better than the kinda mysterious `49 + rttr::enum_cast(bq)`